### PR TITLE
Edit story by story

### PR DESCRIPTION
**Pull request by agallant121**
Edit views and tests to limit functionality based on which user is logged in

### CODE REVIEW

File: app/views/items/show.html.erb [hunk 1]
```diff
@@ -1,58 +1,70 @@
...
+      </p>
       <% if @item.active? %>
-        <p style= "color:green">Active</p>
+        <p style="color:green">Active</p>
       <% else %>
-        <p style= "color:red">Inactive</p>
+        <p style="color:red">Inactive</p>
       <% end %>
+      <% if current_user %>
+      <ul><% if current_user.merchant_id == @item.merchant.id || current_admin %></ul>
```
Comment (agallant121) [inline app/views/items/show.html.erb:28]: Add logic to make sure correct user has the functionality of the view

File: app/views/merchants/show.html.erb [hunk 2]
```diff
@@ -1,16 +1,24 @@
 <h1><%= link_to @merchant.name, "/merchants/#{@merchant.id}" %></h1>
 <p class="address"><%= @merchant.address %></p>
-<p class="address"><%= @merchant.city %>, <%= @merchant.state %> <%= @merchant.zip %></p>
+<p class="address"><%= @merchant.city %>,
+  <%= @merchant.state %>
+  <%= @merchant.zip %></p>
 
 <ul><%= link_to "All #{@merchant.name} Items", "/merchants/#{@merchant.id}/items" %></ul>
+<% if current_user %>
+<ul><% if current_user.merchant_id == @merchant.id || current_admin %></ul>
```
Comment (agallant121) [inline app/views/merchants/show.html.erb:9]: Add logic to make sure correct user has the functionality of the view

File: db/seeds.rb [hunk 3]
```diff
@@ -12,6 +12,8 @@
 #merchants
 bike_shop = Merchant.create(name: "Meg's Bike Shop", address: '123 Bike Rd.', city: 'Denver', state: 'CO', zip: 80203)
 dog_shop = Merchant.create(name: "Brian's Dog Shop", address: '125 Doggo St.', city: 'Denver', state: 'CO', zip: 80210)
+merchant_user = dog_shop.users.create!(name: "Ben", street_address: "891 Penn St. Denver, CO",
```
Comment (agallant121) [inline db/seeds.rb:15]: Make sure user belongs to shop as an emplyee

File: spec/features/items/destroy_spec.rb [hunk 4]
```diff
@@ -4,23 +4,44 @@
   describe 'when I visit an item show page' do
     it 'I can delete an item' do
       bike_shop = Merchant.create(name: "Brian's Bike Shop", address: '123 Bike Rd.', city: 'Denver', state: 'CO', zip: 80203)
+      merchant_user = bike_shop.users.create!(name: "Ben", street_address: "891 Penn St. Denver, CO",
+                                city: "denver",state: "CO",zip: "80206",email: "merchant@gmail.com",password: "hamburger2", role: 2)
```
Comment (agallant121) [inline spec/features/items/destroy_spec.rb:8]: Make sure user is an employee of the specific shop

File: spec/features/items/destroy_spec.rb [hunk 5]
```diff
@@ -4,23 +4,44 @@
...
+      click_link "Delete Item"
 
       expect(current_path).to eq("/items")
       expect("item-#{chain.id}").to be_present
     end
 
     it 'I can delete items and it deletes reviews' do
       bike_shop = Merchant.create(name: "Brian's Bike Shop", address: '123 Bike Rd.', city: 'Denver', state: 'CO', zip: 80203)
+      merchant_user = bike_shop.users.create!(name: "Ben", street_address: "891 Penn St. Denver, CO",
+                                city: "denver",state: "CO",zip: "80206",email: "merchant@gmail.com",password: "hamburger2", role: 2)
```
Comment (agallant121) [inline spec/features/items/destroy_spec.rb:32]: Make sure user is an employee of the specific shop

File: spec/features/items/edit_spec.rb [hunk 6]
```diff
@@ -4,8 +4,18 @@
   describe "When I visit an Item Show Page" do
     describe "and click on edit item" do
       it 'I can see the prepopulated fields of that item' do
+
         @meg = Merchant.create(name: "Meg's Bike Shop", address: '123 Bike Rd.', city: 'Denver', state: 'CO', zip: 80203)
         @tire = @meg.items.create(name: "Gatorskins", description: "They'll never pop!", price: 100, image: "https://www.rei.com/media/4e1f5b05-27ef-4267-bb9a-14e35935f218?size=784x588", inventory: 12)
+        @merchant_user = @meg.users.create!(name: "Ben", street_address: "891 Penn St. Denver, CO",
+                                  city: "denver",state: "CO",zip: "80206",email: "merchant@gmail.com",password: "hamburger2", role: 2)
+        visit '/'
```
Comment (agallant121) [inline spec/features/items/edit_spec.rb:12]: Make sure user is an employee of the specific shop

File: spec/features/items/edit_spec.rb [hunk 7]
```diff
@@ -52,7 +71,16 @@
       it 'I get a flash message if entire form is not filled out' do
         @meg = Merchant.create(name: "Meg's Bike Shop", address: '123 Bike Rd.', city: 'Denver', state: 'CO', zip: 80203)
         @tire = @meg.items.create(name: "Gatorskins", description: "They'll never pop!", price: 100, image: "https://www.rei.com/media/4e1f5b05-27ef-4267-bb9a-14e35935f218?size=784x588", inventory: 12)
-
+        @merchant_user = @meg.users.create!(name: "Ben", street_address: "891 Penn St. Denver, CO",
+                                  city: "denver",state: "CO",zip: "80206",email: "merchant@gmail.com",password: "hamburger2", role: 2)
```
Comment (agallant121) [inline spec/features/items/edit_spec.rb:75]: Make sure user is an employee of the specific shop

File: spec/features/items/new_spec.rb [hunk 8]
```diff
@@ -4,6 +4,15 @@
   describe "When I visit the merchant items index page" do
     before(:each) do
       @brian = Merchant.create(name: "Brian's Dog Shop", address: '125 Doggo St.', city: 'Denver', state: 'CO', zip: 80210)
+      @merchant_user = @brian.users.create!(name: "Ben", street_address: "891 Penn St. Denver, CO",
+                                city: "denver",state: "CO",zip: "80206",email: "merchant@gmail.com",password: "hamburger2", role: 2)
```
Comment (agallant121) [inline spec/features/items/new_spec.rb:8]: Make sure user is an employee of the specific shop

File: spec/features/merchants/destroy_spec.rb [hunk 9]
```diff
@@ -4,8 +4,17 @@
   describe "When I visit a merchant show page" do
     it "I can delete a merchant" do
       bike_shop = Merchant.create(name: "Brian's Bike Shop", address: '123 Bike Rd.', city: 'Richmond', state: 'VA', zip: 80203)
+      admin_user = User.create!(name: "Ben", street_address: "891 Penn St. Denver, CO",
+                                city: "denver",state: "CO",zip: "80206",email: "merchant@gmail.com",password: "hamburger2", role: 3)
```
Comment (agallant121) [inline spec/features/merchants/destroy_spec.rb:8]: Sign in as admin to perform functionality

File: spec/features/merchants/destroy_spec.rb [hunk 10]
```diff
@@ -16,8 +25,17 @@
     it "I can delete a merchant that has items" do
       bike_shop = Merchant.create(name: "Brian's Bike Shop", address: '123 Bike Rd.', city: 'Denver', state: 'CO', zip: 80203)
       chain = bike_shop.items.create(name: "Chain", description: "It'll never break!", price: 50, image: "https://www.rei.com/media/b61d1379-ec0e-4760-9247-57ef971af0ad?size=784x588", inventory: 5)
+      admin_user = User.create!(name: "Ben", street_address: "891 Penn St. Denver, CO",
+                                city: "denver",state: "CO",zip: "80206",email: "merchant@gmail.com",password: "hamburger2", role: 3)
```
Comment (agallant121) [inline spec/features/merchants/destroy_spec.rb:29]: Sign in as admin to perform functionality

File: spec/features/merchants/edit_spec.rb [hunk 11]
```diff
@@ -4,7 +4,17 @@
   describe "After visiting a merchants show page and clicking on updating that merchant" do
     before :each do
       @bike_shop = Merchant.create(name: "Brian's Bike Shop", address: '123 Bike Rd.', city: 'Richmond', state: 'VA', zip: 11234)
+      @merchant_user = @bike_shop.users.create!(name: "Ben", street_address: "891 Penn St. Denver, CO",
+                                city: "denver",state: "CO",zip: "80206",email: "merchant@gmail.com",password: "hamburger2", role: 2)
```
Comment (agallant121) [inline spec/features/merchants/edit_spec.rb:8]: Sign in as Admin to perform functionality

**Review by palworth — APPROVED** [review]
Ready for Master.

**Pull request merged**